lib/pages/login/login_page.dart

```diff
@@ -129,34 +193,36 @@ class LoginPage extends StatelessWidget {
               ),
             ),
             child: Text(AppLocalizations.of(context)!.loginText),
-            onPressed: () {
-              // TODO login https://github.com/luongvo/flutter-survey/issues/7
-              Navigator.of(context).pushNamed(Routes.HOME_PAGE);
-            },
+            onPressed: () => _attemptLogin(),
           ),
         ),
       ],
     );
   }
 
-  InputDecoration _formInputDecoration(BuildContext context, String hint) =>
-      InputDecoration(
-        border: OutlineInputBorder(
-          borderSide: BorderSide.none,
-          borderRadius: BorderRadius.circular(
-            Dimens.inputBorderRadius,
-          ),
-        ),
-        contentPadding: const EdgeInsets.symmetric(
-          horizontal: Dimens.inputHorizontalPadding,
-          vertical: Dimens.inputVerticalPadding,
-        ),
-        fillColor: ColorName.whiteAlpha18,
-        filled: true,
-        hintStyle: Theme.of(context)
-            .textTheme
-            .bodyText1!
-            .copyWith(color: ColorName.whiteAlpha18),
-        hintText: hint,
-      );
+  String? _emailValidator(String? value) {
+    if (value == null || !EmailValidator.validate(value)) {
+      return AppLocalizations.of(context)!.validationErrorEmailInvalid;
+    }
+    return null;
+  }
+
+  String? _passwordValidator(String? value) {
+    if (value == null || value.length < PASSWORD_LENGTH_MIN) {
+      return AppLocalizations.of(context)!.validationErrorEmailInvalid;
+    }
+    return null;
+  }
+
+  Future<void> _attemptLogin() async {
+    if (_formKey.currentState!.validate()) {
+      KeyboardUtil.hideKeyboard(context);
+
+      final LoginModel loginModel =
+          context.read<LoginModel>(loginModelProvider.notifier);
+      loginModel.login(_emailController.text, _passwordController.text);
+    } else {
+      return null;
```

Comment:
do we really need to return here? 🤔  even if we do, I don't think we need the `else` as the return type is always expected as void

Reply:
dart's mystery lol, thanks

Reply:
Fixed https://github.com/luongvo/flutter-survey/pull/51/commits/9f123f22f896223a90f5fb92a347386bd6d970fc